src/platform/packages/private/kbn-index-editor/src/index_update_service.ts

```diff
@@ -383,8 +375,9 @@ export class IndexUpdateService {
           })
         )
         .subscribe({
-          next: ({ updates, response, rows, dataView }) => {
-            // TODO do we need to re-fetch docs using _mget, in order to retrieve a full doc update?
+          next: async ({ updates, response, rows, dataView }) => {
+            // Refresh the data view fields to get new columns types if any
+            await this.data.dataViews.refreshFields(dataView, false, true);
```

Comment:
- I don't think it's a good idea to have an async subscribe callback, did you consider using `switchMap` instead? 
- Also with the latest requirements it might be an overkill as the flyout should be closed automatically after saving